kamon-play-2.6.x/src/main/scala/kamon/play/instrumentation/RequestHandlerInstrumentation.scala

```diff
@@ -15,48 +15,54 @@
 
 package kamon.play.instrumentation
 
-import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
 import kamon.Kamon
 import kamon.context.Context
-import kamon.play.OperationNameFilter
 import kamon.trace.Span
 import kamon.util.CallingThreadExecutionContext
-import org.aspectj.lang.ProceedingJoinPoint
-import org.aspectj.lang.annotation._
-import play.api.mvc.EssentialFilter
 
 import scala.concurrent.Future
 
-@Aspect
-class RequestHandlerInstrumentation {
+trait GenericRequest {
+  val headers: Map[String, String]
+  val method: String
+  val url: String
+  val spanKind: String
+}
+
+trait GenericResponse {
+  def statusCode: Int
+  def reason: String
+}
 
-  private lazy val filter: EssentialFilter = new OperationNameFilter()
+trait GenericResponseBuilder[T] {
+  def build(response: T): GenericResponse
+}
 
-  @Around("execution(* play.core.server.AkkaHttpServer.handleRequest(..)) && args(request, *)")
-  def routeRequestNumberTwo(pjp: ProceedingJoinPoint, request: HttpRequest): Any = {
-    val incomingContext = decodeContext(request)
+object RequestHandlerInstrumentation {
+
+  def handleRequest[T](responseInvocation: => Future[T], request: GenericRequest)(implicit builder: GenericResponseBuilder[T]): Future[T] = {
+    val incomingContext = context(request.headers)
     val serverSpan = Kamon.buildSpan("unknown-operation")
       .asChildOf(incomingContext.get(Span.ContextKey))
       .withMetricTag("span.kind", "server")
-      .withTag("component", "play.server")
-      .withTag("http.method", request.method.value)
-      .withTag("http.url", request.getUri.toString)
+      .withTag("component", request.spanKind)
+      .withTag("http.method", request.method)
```

Comment:
`component` and `http.method` should be metric tags as well. Please take a look at https://github.com/kamon-io/Kamon/issues/525 as well, these little details should be taken into account there!

Reply:
Yes, `component` and `http.method` should be metric tag as well.